clang/lib/Sema/SemaDeclAttr.cpp

```diff
@@ -8314,6 +8350,10 @@ void Sema::ProcessDeclAttributeList(Scope *S, Decl *D,
         Diag(D->getLocation(), diag::err_opencl_kernel_attr) << A;
         D->setInvalidDecl();
       }
+    } else if (const auto *A =
+                   D->getAttr<SYCLIntelSchedulerTargetFmaxMhzAttr>()) {
+      Diag(D->getLocation(), diag::err_opencl_kernel_attr) << A;
+      D->setInvalidDecl();
```

Comment:
This thing is weird, because
- I don't see any test for this diagnostic, its message says `attribute %0 can only be applied to an OpenCL kernel function`, I can't find it in your patch.
- Tests and doc seems like we should allow it for non-OpenCL kernel function. BTW, I'm not sure that there is a way to apply an attribute to OpenCL kernel in *SYCL* at all.

So, this code is weird, because I'm not sure that it looks correct from logic point of view and it seems like it doesn't work anyway.

Reply:
Alright I got it. The whole bunch of such errors is disabled of sycl host and device compilation on line 8292 in SemaDeclAttr.cpp file, so even additional check on line 8313 is not needed.
And you can remove check that you added, because it makes sense only for OpenCL and your attribute seems to be enabled only for SYCL.

Reply:
Right. Removing it per this finding and offline discussion.

Reply:
Line 8313 is `// FIXME: We should be able to handle these cases in TableGen.` 
May be you meant something else, but I think I understand what you mean by your comment, overall.